src/commonMain/kotlin/com/algolia/search/configuration/HttpClient.kt

```diff
@@ -44,11 +47,25 @@ internal fun HttpClientConfig<*>.configure(configuration: Configuration) {
         logger = Logger.SIMPLE
     }
     install(UserAgent) {
-        agent = "Algolia for Kotlin (${BuildConfig.version})"
+        agent =  "Algolia for Kotlin (${BuildConfig.version})"
     }
-    configuration.defaultHeaders?.let {
-        install(DefaultRequest) {
+    defaultRequest {
+        configuration.defaultHeaders?.let {
             it.forEach { (key, value) -> header(key, value) }
         }
+        if (method.canCompress()) {
+            compressionHeader(configuration.compression)
+        }
+    }
+}
+
+internal fun HttpMethod.canCompress(): Boolean {
+    return this == HttpMethod.Post || this == HttpMethod.Put
+}
+
+internal fun HttpRequestBuilder.compressionHeader(compression: Compression) {
+    when (compression) {
+        Compression.Gzip -> header(HttpHeaders.ContentEncoding, "gzip")
+        else -> Unit
     }
 }
```

Comment:
nit: could be nice to open another PRs to have all files finishing with a new line.

Reply:
Do you think it would be useful to have a new line at the end of each file ?

Reply:
It's a [posix standard convention ](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline)I have seen in many projects. It's up to you :)

Reply:
I might do it then :) Thanks